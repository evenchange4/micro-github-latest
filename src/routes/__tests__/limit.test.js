/* eslint global-require: 0, no-throw-literal: 0 */

beforeEach(() => {
  jest.resetModules();
  jest.mock('micro', () => ({
    send: (res, code, content) => ({ res, code, content }),
  }));
});

it('should return limit', async () => {
  jest.doMock('../../utils/API', () => ({
    getRateLimit: () => ({}),
  }));

  const limit = require('../limit');

  expect(await limit()).toMatchSnapshot();
});

it('should return limit with error', async () => {
  jest.doMock('../../utils/API', () => ({
    getRateLimit: () => {
      throw new Error('error');
    },
  }));

  const limit = require('../limit');

  expect(await limit()).toMatchSnapshot();
});
