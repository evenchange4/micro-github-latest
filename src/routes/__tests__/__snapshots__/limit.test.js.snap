// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return limit 1`] = `
Object {
  "code": 200,
  "content": Object {},
  "res": undefined,
}
`;

exports[`should return limit with error 1`] = `
Object {
  "code": 404,
  "content": [Error: error],
  "res": undefined,
}
`;
